cli/internal/runcache/runcache.go

```diff
@@ -66,77 +54,59 @@ func AddFlags(opts *Opts, flags *pflag.FlagSet) {
 	}
 }
 
-var _outputModeHelp = `Set type of process output logging. Use full to show
-all output. Use hash-only to show only turbo-computed
-task hashes. Use new-only to show only new output with
-only hashes for cached tasks. Use none to hide process
-output.`
-
-type logsModeValue struct {
+type taskOutputModeValue struct {
 	opts *Opts
 }
 
-func (l *logsModeValue) String() string {
-	if l.opts.CacheHitLogsMode == FullLogs && l.opts.CacheMissLogsMode == FullLogs {
-		return logsModeFull
-	} else if l.opts.CacheHitLogsMode == NoLogs && l.opts.CacheMissLogsMode == NoLogs {
-		return logsModeNone
-	} else if l.opts.CacheHitLogsMode == HashLogs && l.opts.CacheMissLogsMode == HashLogs {
-		return logsModeHashOnly
-	} else if l.opts.CacheHitLogsMode == HashLogs && l.opts.CacheMissLogsMode == FullLogs {
-		return logsModeNewOnly
-	} else {
-		panic(fmt.Sprintf("Invalid output logs mode. Hit %v, miss %v", l.opts.CacheHitLogsMode, l.opts.CacheMissLogsMode))
-	}
+func (l *taskOutputModeValue) String() string {
+	return l.opts.TaskOutputModeOverride
 }
 
-func (l *logsModeValue) Set(value string) error {
-	switch value {
-	case logsModeFull:
-		l.opts.CacheMissLogsMode = FullLogs
-		l.opts.CacheHitLogsMode = FullLogs
-	case logsModeNone:
-		l.opts.CacheMissLogsMode = NoLogs
-		l.opts.CacheHitLogsMode = NoLogs
-	case logsModeHashOnly:
-		l.opts.CacheMissLogsMode = HashLogs
-		l.opts.CacheHitLogsMode = HashLogs
-	case logsModeNewOnly:
-		l.opts.CacheMissLogsMode = FullLogs
-		l.opts.CacheHitLogsMode = HashLogs
-	default:
-		return fmt.Errorf("unknown output-mode: %v", value)
+func (l *taskOutputModeValue) Set(value string) error {
+	// Validate the output mode before allowing it to be set
+	_, err := util.FromTaskOutputModeString(value)
+	if err != nil {
+		return fmt.Errorf("must be one of \"%v\"", l.Type())
 	}
+	l.opts.TaskOutputModeOverride = value
 	return nil
 }
 
-func (l *logsModeValue) Type() string {
-	return "full|none|hash-only|new-only"
+func (l *taskOutputModeValue) Type() string {
```

Comment:
Honestly, this feels a bit excessive. We're already naming the strings in the description for the command, we should assume any new options added to the enum will be added here as a matter of consistency.

Reply:
I'll leave it. Even if it's missed in the description for some reason, it's better to document that the option exists rather than miss the fact that it does.